salt/hostname/init.sls

```diff
@@ -0,0 +1,11 @@
+{% set hostname = grains['id'] + '.' + pillar['internal_infra_domain'] %}
+
+/etc/hostname:
```

Comment:
As it turns out, no, it doesn't. (I thought so too, but it didn't actually do it in my testing)